core/IssueQueue.cpp

```diff
@@ -101,29 +100,66 @@ namespace olympia
         // FIXME: Now every source operand should be ready
         auto reg_file = core_types::RegFile::RF_INTEGER;
         const auto srcs = ex_inst->getRenameData().getSourceList();
-        if (srcs.size() > 0)
-        {
-            reg_file = srcs[0].rf;
-        }
-        const auto & src_bits = ex_inst->getSrcRegisterBitMask(reg_file);
-        if (scoreboard_views_[reg_file]->isSet(src_bits))
-        {
-            // Insert at the end if we are doing in order issue or if the scheduler is
-            // empty
-            ILOG("Sending to issue queue " << ex_inst);
-            // will insert based on if in_order_issue_ is set
-            // if it is, will be first in first out, if not it'll be by age, so by UniqueID (UID)
-            ready_queue_.insert(ex_inst);
-            ev_issue_ready_inst_.schedule(sparta::Clock::Cycle(0));
+        
+        if(srcs.size() > 1 && srcs[0].rf != srcs[1].rf && ex_inst->isVector()){
```

Comment:
I would had a method to the instruction class for this check instead of doing it here. Like `isV2S` and `isS2V`.

Reply:
Added in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)